uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/pageObjects/Page.java

```diff
@@ -49,4 +50,11 @@ private void clickLogout() {
     public void clearCookies() {
         driver.manage().deleteAllCookies();
     }
+
+    public static void validateUrlStartsWithWait(WebDriver driver, String currentUrlStart) throws InterruptedException {
+        if (!driver.getCurrentUrl().startsWith("https://www.google.com")) {
```

Comment:
Is this line supposed to be `if (!driver.getCurrentUrl().startsWith(currentUrlStart)) {`?

Reply:
Yes because if the SAML login takes too long then the current URL is not the expected one.

Therefore ,wait for 5 sec. And then, if still not expected one then fail

Reply:
I see. The current code in this PR is hardcoded to expect `https://www.google.com` though. I was just asking if the code should expect the `currentUrlStart` from the function param.

Reply:
ups... sorry, now I see it. Yes I tested many times and behavior in github action and concourse pipeline was always different so.... these waits should help , but should be only done the described situations

Thanks for the hint